science/imager/Portfile

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           compilers 1.0
+PortGroup           active_variants 1.1
+
+name                imager
+version             202211
+set my_version      [string tolower [clock format [clock scan 2000-[string range ${version} 4 5]-10] -format %b]][string range ${version} 2 3][string range ${version} 6 end]
+supported_archs     arm64 x86_64
+set my_machine      [if {[string match *arm* ${os.arch}]} {list arm64} {list x86_64}]
+categories          science
+platforms           darwin
+maintainers         {u-bordeaux.fr:julien.penguen @penguen}
+license             permissive
+
+description         Radio-interferometry data analysis software
+
+long_description    IMAGER (and its associated visualization tool VIEWER) is a state-of-the-art \
+                    software oriented toward (sub-)millimeter radioastronomical interferometry. \ 
+                    It is used to reduce data acquired with the IRAM NOEMA interferometer \
+                    and the ALMA or JVLA arrays, with a simple, convenient user interface.
+
+homepage            https://imager.oasu.u-bordeaux.fr
+master_sites        https://nuage.oasu.u-bordeaux.fr/index.php/s/PoxdxMTTYXqNwKF \
+                    https://nuage.oasu.u-bordeaux.fr/index.php/s/AjgLDY86zX65JjL
+distname            ${name}-${my_version}
+use_xz              yes
+
+checksums           rmd160  9c1185a5c5e9fc54612808977ee8f548b2258d31 \
+                    sha256  c5d0288e3e95700655219d2617a3f109ffc2a37fbab983877c3c73410c7df952 \
+                    size    60725540
+
+python.version     311
+
+patch.pre_args      -p1
+patchfiles          patch-admin-Makefile.def.diff \
+                    patch-admin-Makefile.python.diff \
+                    patch-admin-Makefile.build.diff \
+                    patch-admin-gildas-env.sh.diff \
+                    patch-admin-define-system.sh.diff
+
+depends_lib         path:lib/pkgconfig/gtk+-2.0.pc:gtk2 \
+                    port:libpng \
+                    port:python${python.version} \
+                    port:py${python.version}-numpy
+
+# coreutils needed to use GNU gsort instead of native BSD sort as older version
+# of sort do not support sort -V:
+depends_build       port:cfitsio \
+                    port:pkgconfig \
+                    path:libexec/coreutils/libstdbuf.so:coreutils
+
+set my_build_opts   "-o openmp"
+#variant openmp description {Add OpenMP support} {set my_build_opts "-o openmp"}
+
+# need x11 and cannot be used with quartz; see ticket #42886
+require_active_variants gtk2 x11 quartz
+
+# need the fortran interface; see ticket #50543
+compilers.enforce_some_fortran cfitsio
+
+compilers.choose    fc
+compilers.setup     -gcc46 require_fortran
+configure.python    ${prefix}/bin/python3.11
```

Comment:
You may want to compute this from `python.version` so that when, every year, the python version is increased, it only has to be done on one line (the `python.version` line).

Reply:
the syntaxes are different between python.version  ( 311) in line 34 and in configure.python (python3.11 required) in line 65

Reply:
> the syntaxes are different between python.version ( 311) in line 34 and in configure.python (python3.11 required) in line 65

This is one pattern we use to deal with that. So replace your current `python.version 311` line with this block:

```
set py_ver          3.11
set py_ver_nodot    [string map {. {}} ${py_ver}]
python.version      ${py_ver_nodot}
```

And then later in the portfile, you can refer to variable `${py_ver}` or `${py_ver_nodot}`, depending on which one you need.